modules/sql-engine/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItComputeSystemViewTest.java

```diff
@@ -0,0 +1,309 @@
+                            new MetadataMatcher().name("STATUS").type(ColumnType.STRING).nullable(true),
+                            new MetadataMatcher().name("CREATE_TIME").type(ColumnType.TIMESTAMP).nullable(true),
+                            new MetadataMatcher().name("START_TIME").type(ColumnType.TIMESTAMP).nullable(true),
+                            new MetadataMatcher().name("FINISH_TIME").type(ColumnType.TIMESTAMP).nullable(true)
+                    )
+                    .check();
+        } finally {
+            execution.cancelAsync();
+
+            await().until(execution::stateAsync, willBe(jobStateWithStatus(CANCELED)));
+
+            closeQuiet(entryNode);
+        }
+    }
+
+    @ParameterizedTest
+    @ValueSource(booleans = {true, false})
+    void viewRunningJobs(boolean isClient) {
+        Ignite entryNode = isClient ? IgniteClient.builder().addresses("localhost").build() : CLUSTER.node(0);
+        Ignite targetNode = CLUSTER.node(0);
+
+        try {
+            ClockService clockService = unwrapIgniteImpl(targetNode).clockService();
+
+            long tsBefore = clockService.now().getPhysical();
+
+            JobDescriptor<Void, Void> job = JobDescriptor.builder(InfiniteJob.class).units(List.of()).build();
+            JobExecution<Void> execution = entryNode.compute().submit(JobTarget.node(clusterNode(targetNode)), job, null);
+
+            await().until(execution::stateAsync, willBe(jobStateWithStatus(EXECUTING)));
+
+            long tsAfter = clockService.now().getPhysical();
+
+            String query = "SELECT * FROM SYSTEM.COMPUTE_TASKS WHERE STATUS = ?";
+
+            List<List<Object>> res = sql(0, query, EXECUTING.name());
+
+            verifyQueryInfo(res.get(0), List.of(targetNode.name()), EXECUTING.name(), tsBefore, tsAfter);
+
+            IgniteTestUtils.await(execution.cancelAsync());
+
+            await().until(execution::stateAsync, willBe(jobStateWithStatus(CANCELED)));
+
+            // Second Job call on different node.
+            job = JobDescriptor.builder(InfiniteJob.class).units(List.of()).build();
+
+            tsBefore = clockService.now().getPhysical();
+
+            targetNode = CLUSTER.node(1);
+
+            execution = entryNode.compute().submit(JobTarget.node(clusterNode(targetNode)), job, null);
+
+            await().until(execution::stateAsync, willBe(jobStateWithStatus(EXECUTING)));
+
+            tsAfter = clockService.now().getPhysical();
+
+            query = "SELECT * FROM SYSTEM.COMPUTE_TASKS WHERE COORDINATOR_NODE_ID = ? AND STATUS = ?";
+
+            res = sql(0, query, targetNode.name(), EXECUTING.name());
+
+            verifyQueryInfo(res.get(0), List.of(targetNode.name()), EXECUTING.name(), tsBefore, tsAfter);
+
+            IgniteTestUtils.await(execution.cancelAsync());
+
+            await().until(execution::stateAsync, willBe(jobStateWithStatus(CANCELED)));
```

Comment:
Since we are not only displaying currently running jobs, maybe we should also check the view data after a job is cancelled?

Reply:
in such a case we need to check inner compute api invariants, i don\`t like it ..